integration/simulation/validate_chain.go

```diff
@@ -240,12 +251,21 @@ func checkBlockchainOfObscuroNode(t *testing.T, node *host.Node, minObscuroHeigh
 }
 
 func extractWithdrawals(node *host.Node) (totalSuccessfullyWithdrawn uint64, numberOfWithdrawalRequests int) {
+	head := node.DB().GetCurrentRollupHead()
+	if head == nil {
+		panic("the current head should not be nil")
+	}
 	// sum all the withdrawals by traversing the node headers from Head to Genesis
-	for r := node.DB().GetCurrentRollupHead(); r != nil; r = node.DB().GetRollupHeader(r.Parent) {
+	for r := head; ; r = node.DB().GetRollupHeader(r.ParentHash) {
+		if r != nil && r.Number == obscurocommon.L1GenesisHeight {
```

Comment:
Should these conditions be in the loop stop definition instead of being in the main body ? (inverted ofc :) )

Reply:
It is much easier to debug this way. I'd leave it like this.